src/main/scala/is/hail/HailContext.scala

```diff
@@ -140,6 +140,25 @@ object HailContext {
       consoleLog.addAppender(new ConsoleAppender(new PatternLayout(HailContext.logFormat), "System.err"))
   }
 
+  def getOrCreate(sc: SparkContext = null,
+    appName: String = "Hail",
+    master: Option[String] = None,
+    local: String = "local[*]",
+    logFile: String = "hail.log",
+    quiet: Boolean = false,
+    append: Boolean = false,
+    minBlockSize: Long = 1L,
+    branchingFactor: Int = 50,
+    tmpDir: String = "/tmp"): HailContext = synchronized {
+
+    if (get != null) {
+      get
```

Comment:
I'm a bit concerned about initializing with different params than the existing context has. any suggestions?

Reply:
Anecdotally, we haven't had any complaints about the similar semantics in `SparkContext.getOrCreate`. I think it's actually more likely that validating the settings by default would cause user problems since minor changes in initialization would cause subsequent calls to fail. For instance, if the first user to create a HailContext on a Databricks cluster used toggles quiet mode, subsequent calls would fail.

A user can validate the returned HailContext if they are sensitive to some of the settings.

Reply:
> A user can validate the returned HailContext

There's no way to do this in 0.2 - we no longer expose HailContext as a user-visible object.

I take your point about throwing errors for small changes in config, but I think at the very least we need to warn. I can PR onto this branch this afternoon when I have some time.

Reply:
Ah of course, I forgot that there's not a scala API 🤦‍♂️. Warning sounds reasonable to me.

Reply:
Ok, I added some warnings.